src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -718,3 +719,55 @@ def causal_mask_input_present(model_path: str) -> bool:
     def _reset_engines_cache(self):
         self.engine.reset_kv_cache()
         self.multitoken_engine.reset_kv_cache() if self.multitoken_engine else None
+
+    def _update_context(self, **kwargs) -> Context:
```

Comment:
A bit hestitant to pack more auxiliary functions into `text_generation.py`. Can we somehow import this as a helper func?

Reply:
could probably move it to helpers if we see multi-stream being an issue in other pipelines.

Reply:
I'd do it now tbh.

Reply:
Done!